lib/internal/url.js

```diff
@@ -932,6 +935,60 @@ function urlToOptions(url) {
   return options;
 }
 
+function getPathFromURLWin32(url) {
+  var hostname = url.hostname;
+  var pathname = url.pathname;
+  if (hostname !== '') {
+    // If hostname is set, then we have a UNC path
+    return `//${hostname}${pathname}`;
+  } else {
+    // Otherwise, it's a local path that requires a drive letter
+    var letter = pathname.codePointAt(1) | 0x20;
+    var sep = pathname[2];
+    if (letter < 97 || letter > 122 ||   // a..z A..Z
+        (sep !== ':' && sep !== '|')) {
+      return new Error('File URLs must specify absolute paths');
+    }
+    pathname = pathname.slice(1);
+    for (var n = 0; n < pathname.length; n++) {
+      if (pathname[n] === '%') {
```

Comment:
There's no need for the extra check. If `n >= pathname.length`, the `pathname[n]` will return `undefined`, which causes this check to fail.